pkg/readiness/ready_tracker.go

```diff
@@ -262,21 +278,27 @@ func (t *Tracker) Satisfied() bool {
 			}
 		}
 	}
-	log.V(1).Info("all expectations satisfied", "tracker", "constraints")
+	log.V(logging.DebugLevel).Info("all expectations satisfied", "tracker", "constraints")
 
 	if !t.config.Satisfied() {
+		log.V(logging.DebugLevel).Info("expectations unsatisfied", "tracker", "config")
 		return false
 	}
 
 	if operations.HasValidationOperations() {
-		configKinds := t.config.kinds()
-		for _, gvk := range configKinds {
+		if !t.syncsets.Satisfied() {
+			log.V(logging.DebugLevel).Info("expectations unsatisfied", "tracker", "syncset")
+			return false
+		}
+
+		for _, gvk := range t.DataGVKs() {
```

Comment:
where do we ensure t.DataGVKs includes both t.config.kinds and syncsets gvks?

Reply:
Good question! It's on lines 798 -- 809 in the `trackSyncSources` func.

```
dt := t.ForData(gvkCpy)
```

we aggregate all the gvks from both Config and SyncSet resources then the underlying `ForData` call will populate "keys" for all gvks that were aggregated.